src/main/scala/chisel3/util/experimental/AutoBlackBox.scala

```diff
@@ -0,0 +1,32 @@
+package chisel3.util.experimental
+
+import chisel3._
+
+import scala.collection.immutable.SeqMap
+
+class AutoBlackBox(
+  val verilog:      String,
+  val signalFilter: String => Boolean = _ => true)
+    extends FixedIOExtModule(
+      ioGenerator = new AutoBundleFromVerilog(
+        SeqMap.from(SlangUtils.verilogModuleIO(SlangUtils.getVerilogAst(verilog)))
+      )(signalFilter)
+    ) {
+  override def desiredName = SlangUtils.verilogModuleName(SlangUtils.getVerilogAst(verilog))
+}
+
+class AutoBundleFromVerilog(
+  allElements:  SeqMap[String, Data]
+)(signalFilter: String => Boolean)
+    extends Record {
+  val elements: SeqMap[String, Data] = allElements.filter(n => signalFilter(n._1)).map {
+    case (k, v) =>
+      k -> chisel3.reflect.DataMirror.internal.chiselTypeClone(v)
+  }
+  def apply(data: String) = elements.getOrElse(
+    data,
+    throw new ChiselException(
+      s"$data not found in Verilog IO: \n ${allElements.keys.mkString("\n")}"
+    )
+  )
+}
```

Comment:
I think it would make sense to have companion objects that take a File or Verilog String (or both!) and materialize the source locator there, then pass that to `AutoBlackBox`.